core/src/cluster_info.rs

```diff
@@ -1400,14 +1404,24 @@ impl ClusterInfo {
             })
             .collect()
     }
-    fn new_push_requests(&self) -> Vec<(SocketAddr, Protocol)> {
+    fn new_push_requests(
+        &self,
+        gossip_validators: Option<&HashSet<Pubkey>>,
+    ) -> Vec<(SocketAddr, Protocol)> {
         let self_id = self.id();
         let (_, push_messages) = self
             .time_gossip_write_lock("new_push_requests", &self.stats.new_push_requests)
             .new_push_messages(timestamp());
         let messages: Vec<_> = push_messages
             .into_iter()
             .filter_map(|(peer, messages)| {
+                if let Some(gossip_validators) = gossip_validators {
```

Comment:
I suspect limiting it in `crds_gossip_push::push_options` and `crds_gossip_pull::pull_options` might be the better/more efficient way. With this you are essentially considering the entire network and then throwing away packets that don't match, but you really could be forming your MST around the nodes that you can actually talk to and sending all your traffic to them.

Consider if you have 200 potential nodes, and only 2 in your trusted set. Your message generation would only be 1% efficient.

@behzadnouri @carllin if they have any opinions as well.

Reply:
Yeah, makes sense.  I got shy about plumbing this into crds_gossip and crds_gossip_push.  I'll channel my inner 🦁 more.

Reply:
limiting in ` crds_gossip_push::push_options` and `crds_gossip_pull::pull_options` makes sense to me!

Reply:
Done